### ISSUE TITLE
OpenSSL conf missing in non-dev Image

### ISSUE THREAD

**Issue by jiw-mh** [brefphp/aws-lambda-layers#154]
It seems like the `openssl.cnf` file is missing from the regular images (`bref/php-83` or `bref/php-83-fpm`) but it is present in the `...-dev` images.
 
```
> docker run -it --entrypoint /bin/sh bref/php-83-fpm-dev -c "ls /opt/bref/ssl/"

cert.pem  ct_log_list.cnf       misc         openssl.cnf.dist
certs     ct_log_list.cnf.dist  openssl.cnf  private
```

```
> docker run -it --entrypoint /bin/sh bref/php-83-fpm -c "ls /opt/bref/ssl/"

cert.pem
```

I noticed that openssl is tested in the line below, but the random bytes method works independently of the configuration. A test like creating private keys might be more reliable?

https://github.com/brefphp/aws-lambda-layers/blob/1ecad625595524d2d926f0e92b34a1b455da348d/tests/test_2_extensions.php#L61

**Comment by mnapoli** [comment]
Sure! I'm interested in the impact on runtimes?

**Comment by jiw-mh** [comment]
Oh, OpenSSL will simply not work without the configuration with some API's. In our case it is the creation of a private key: 

```
docker run -it --entrypoint /opt/bin/php bref/php-83-fpm-dev -r "var_dump(openssl_pkey_new(['digest_alg' => 'RS256', 'private_key_bits' => 2048, 'private_key_type' => OPENSSL_KEYTYPE_RSA ]));"
```

The `-dev` images will return a key, while the regular ones will return `false`.

**Comment by mnapoli** [comment]
Understood, thank you for clarifying!

We might need to copy more files here: https://github.com/brefphp/aws-lambda-layers/blob/main/php-83/Dockerfile#L475-L476

Would you be able to send a pull request?

**Comment by jiw-mh** [comment]
I am not sure I have the capability to test a PR if I were to send it. Aside from this, I noticed I forgot to mention that this problem exists for php-81 and 82 as well (haven't tested 80)

Now that I looked a bit more into the Dockerfile, I am puzzled why it works in `-dev` image. 

https://github.com/brefphp/aws-lambda-layers/blob/f9a9907699b43ede302b1fcf88f2bbdccf67ec50/layers/fpm-dev/Dockerfile#L23

It seems like the `-dev` image is based of the other images and I can't see what after that adds the missing files?! Maybe it would result in a better fix?

**Comment by mnapoli** [comment]
Oh good point, no idea either right now 🤔

**Comment by alexw23** [comment]
The issue is that -dev references `build_extension`, which is an alias for build-php (aka build_environment in main Dockerfile):
https://github.com/brefphp/aws-lambda-layers/blob/62d9366f1c5e81bdf281df389d0172069502a09a/layers/fpm-dev/Dockerfile#L6

Which then further down the entire `/opt` is copied:
https://github.com/brefphp/aws-lambda-layers/blob/62d9366f1c5e81bdf281df389d0172069502a09a/layers/fpm-dev/Dockerfile#L25

However in the main php build (which is a reference in the build target [function](https://github.com/brefphp/aws-lambda-layers/blob/main/docker-bake.hcl#L37) which is based on `isolation`, only /bref-layer is being copied:
https://github.com/brefphp/aws-lambda-layers/blob/62d9366f1c5e81bdf281df389d0172069502a09a/php-83/Dockerfile#L502-L505

But in `build_environment` the `ssl/openssl.cnf is` never copied to to /bref-layer, only the CA_BUNDLE:
https://github.com/brefphp/aws-lambda-layers/blob/62d9366f1c5e81bdf281df389d0172069502a09a/php-83/Dockerfile#L497

Fixed it in my PR #172 by adding the following in `build_enviroment`
https://github.com/alexw23/aws-lambda-layers/blob/185ae0e69d00f88c620df51424ef24186066b313/php-83/Dockerfile#L500
